### PR TITLE
[graviton][sagemaker][test] SageMaker "Local" tests update for Graviton

### PR DESCRIPTION
**Pull request by xncqr**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Updated the constant variable names for the ARM ami_ids as the OS is Ubuntu 20.04 over 18.04. Updated SageMaker "local" tests with a helper to select the proper AMI id for testing a docker image. This was done to support Graviton (ARM) Docker images for SageMaker.

Removed pinning for "torchserve" and "numpy" in PyTorch SM test requirements as they were having dependencies with the new pip. Ran the regular SM testing for x86_64 to verify testing is not harmed with normal SM DLCs.

The Graviton SM image builds are commented (disabled) as they would not pass the remote SM tests till the production endpoints understand graviton (arm64) instances.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/sagemaker.py [hunk 1]
```diff
@@ -224,7 +245,7 @@ def install_sm_local_dependencies(framework, job_type, image, ec2_conn, ec2_inst
     python_invoker = get_python_invoker(ec2_instance_ami)
     # Install custom packages which need to be latest version"
     # using virtualenv to avoid package conflicts with the current packages
-    ec2_conn.run(f"sudo apt-get install virtualenv -y ")
+    ec2_conn.run(f"sudo apt-get install virtualenv python3-pip -y ")
```
Comment (saimidu) [inline test/test_utils/sagemaker.py:248]: Please check that `pip` doesn't already exist before installing system pip, because it will conflict with version that might already be installed, and cause test errors on non Graviton AMIs.

File: test/test_utils/sagemaker.py [hunk 2]
```diff
@@ -53,14 +57,31 @@ def assign_sagemaker_remote_job_instance_type(image):
...
+        return "c6g.2xlarge"
+    elif "tensorflow" in image and "inference" in image and "gpu" in image:
         return "g4dn.xlarge"
     elif "autogluon" in image and "gpu" in image:
         return "p3.2xlarge"
     elif "trcomp" in image:
         return "p3.2xlarge"
     return "p3.8xlarge" if "gpu" in image else "c5.18xlarge"
 
+def assign_sagemaker_ami(image, region):
```
Comment (saimidu) [inline test/test_utils/sagemaker.py:70]: Please clarify that this is the AMI for local tests.

Perhaps
```
def assign_sagemaker_local_test_ami(...)
```
?

File: test/test_utils/sagemaker.py (hunk 1 above)
Comment (xncqr) [reply test/test_utils/sagemaker.py:248]: it doesn't as the process was failing with "no pip found", being the reason for the add.

File: test/test_utils/sagemaker.py (hunk 2 above)
Comment (xncqr) [reply test/test_utils/sagemaker.py]: Updated

**Review by saimidu — APPROVED** [review]
